.github/workflows/compile.yml

```diff
@@ -24,67 +24,25 @@ jobs:
           # Git history is needed for `git describe` in the build to work.
           fetch-depth: 0
 
-      - name: Install packages (Linux)
-        if: startsWith(matrix.os, 'ubuntu-')
-        run: |
-          sudo apt install -y --no-install-recommends pkg-config libgmp-dev curl ninja-build
-
-      - name: Install packages (macOS)
-        if: startsWith(matrix.os, 'macos-')
-        run: |
-          brew install opam llvm lld
-          echo "$(brew --prefix llvm)/bin" >> $GITHUB_PATH
-
-      - name: Install CMake
-        uses: jwlawson/actions-setup-cmake@v2
+      - name: Common setup
+        uses: ./.github/actions/sail-setup
         with:
           cmake-version: "${{ matrix.cmake_version }}"
-
-      - name: Install Sail (Linux)
-        if: startsWith(matrix.os, 'ubuntu-')
-        run: |
-          sudo mkdir -p /usr/local
-          curl --location https://github.com/rems-project/sail/releases/download/$SAIL_VERSION-linux-binary/sail.tar.gz | sudo tar xvz --directory=/usr/local --strip-components=1
-
-      - name: Restore cached opam (macOS)
-        if: startsWith(matrix.os, 'macos-')
-        id: cache-opam-restore
-        uses: actions/cache/restore@v4
-        with:
-          path: ~/.opam
-          key: ${{ matrix.os }}-${{ matrix.cmake_version }}-${{ env.SAIL_VERSION }}-opam
-
-      - name: Install Sail (macOS)
-        if: steps.cache-opam-restore.outputs.cache-hit != 'true' && startsWith(matrix.os, 'macos-')
-        run: |
-          opam init --auto-setup --bare
-          opam switch create default 5.3.0
-          eval $(opam env)
-          opam update
-          opam install -y sail.$SAIL_VERSION
-
-      - name: Save cached opam (macOS)
-        if: startsWith(matrix.os, 'macos-')
-        id: cache-opam-save
-        uses: actions/cache/save@v4
-        with:
-          path: ~/.opam
-          key: ${{ steps.cache-opam-restore.outputs.cache-primary-key }}
-
-      - name: Load OPAM environment (macOS)
-        if: startsWith(matrix.os, 'macos-' )
-        run: echo "$HOME/.opam/default/bin" >> $GITHUB_PATH
+          sail-version: "${{ env.SAIL_VERSION }}"
```

Comment:
Could add this as the default to action.yml to avoid needing to update it in two places?